### ISSUE TITLE
Error:  unable to open output file 'main.o': 'Permission denied'

### ISSUE THREAD

**Issue by boazsegev** [hebaishi/linter-gcc#19]
Linter-gcc fails to work for my C project. Instead, a single error is presented: 

      unable to open output file 'main.o': 'Permission denied'

Environment: Atom on MacOS X 10.11.2 (latest). Linter-Gcc v. 0.5.10

Log (from the Dev console):

> linter-gcc trying: /Users/USER/rC/proj/src/.gcc-flags.json

> /Users/USER/.atom/packages/linter-gcc/lib/config.js:33 linter-gcc trying: /Users/USER/rC/proj/.gcc-flags.json

> /Users/USER/.atom/packages/linter-gcc/lib/main.js:124 linter-gcc: gcc -c -g -Wall -O3 -fmax-errors=0 -I/Users/USER/rC/proj -I/Users/USER/rC/proj/lib -I/Users/USER/rC/proj/tmp /Users/USER/rC/proj/src/main.c

**Comment by hebaishi** [comment]
Could you try removing the ```-c``` in the C flags option on the linter-gcc configuration page? If that fixes the issue for you, I will post an update informing OSX users about this problem.

**Comment by boazsegev** [comment]
I've tried removing the `-c`, as well as using `-S`, both failed.

I continued playing with this a bit. I have it mostly working by using the flag:

 `-Wall -c -o /Useres/USER/path/to/custom/tmp/folder/tmp.o`

But... I'm not sure it's working as expected... When there aren't any errors, I get an error message stating `error=0'`... Also, some errors claimed by linter are not errors at all (structs aren't recognized as well as might be expected)... maybe it's using C++ while I'm using C (keyword collision)... I'm not sure.

**Comment by boazsegev** [comment]
Another optional solution that has the same effect (shows errors, including the reservations I already stated) is by setting a no-output file using the flag: `-fsyntax-only`

**Comment by boazsegev** [comment]
Update:

using the `-fsyntax-only` flag, only the `errors=0'` issue remains (getting this strange error line instead of getting an OK green happy marker.

**Comment by hebaishi** [comment]
This seems like a clang-related issue. See [here](https://github.com/hebaishi/linter-gcc/issues/18) for more information. You could try switching to [linter-clang](https://atom.io/packages/linter-clang) and seeing if the issue persists.

**Comment by boazsegev** [comment]
Hussar,

Thank you for taking the time with this. Our community is lucky to have you.

I assume the issue can be closed by adding the `-fsyntax-only` flag...? - or maybe it was all clang related? I'm sorry I caused you any bother.

**Comment by hebaishi** [comment]
Thank you very much for your kind comment. Yes, I do think this issue is clang-related. Unfortunately, the gcc/g++ commands on OSX both link to clang. See [this](http://stackoverflow.com/questions/19535422/os-x-10-9-gcc-links-to-clang) stackoverflow post for more information.

**Comment by Foadsf** [comment]
adding the -fsyntax-only flag removed the error. but I still have no idea why it happens. I had already redirected the gcc to my brew installed gcc-6 and had removed the -c flag.

**Comment by cristiano-c** [comment]
> [https://github.com/hebaishi/linter-gcc/issues/19#issuecomment-168113303](boazsegev commented on 31 Dec 2015)

I have same issue.

Please, can someone explain me how to change GCC linter arguments?

**Comment by aristizabal95** [comment]
@cristiano-c I'm a little late to the party, but had the same issue and fixed it as proposed here

You have to edit the file `~/.atom/packages/linter-gcc2/lib/main.js` to add `-fsyntax-only` inside the cpp default flags.

This is how it should look before, around line `24`
```javascript
gccDefaultCppFlags: {
  title: "C++ Flags",
  description: "Supports the use of escaped characters",
  type: "string",
  default: "-c -fsyntax-only -Wall -std=c++11",
  order : 3
},
```
